src/contributions/sites/SitePopup.js

```diff
@@ -0,0 +1,199 @@
+            setTown(selectedSite["town"]);
+            setSiteName(selectedSite["site_name"]);
+        }
+        else {
+            setCounty("");
+            setTown("");
+            setSiteName("");
+        }
+    }, [selectedSite]);
+
+    useEffect(() => {
+        console.log("CustomPopup::useEffect selectedSite", selectedSite);
+        if (selectedSite) {
+            setSubmitDisabled(
+                selectedSite["county"] === county
+                && selectedSite["town"] === town
+                && selectedSite["site_name"] === siteName
+                && selectedSite["street"] === street
+                && selectedSite["zipcode"] === zipcode
+                && selectedSite["lat"] === lat
+                && selectedSite["long"] === long);
+        }
+        else {
+            setSubmitDisabled(!county || !town || !siteName);
+        }
+    }, [selectedSite, county, town, siteName, street, zipcode, lat, long]);
+
+    return(
+        <div>
+        <Popup
+            show={show}
+            onHide={onClose}
+            title={(selectedSite ? "Update" : "Add")
+                    + " Site "}
+            body={
+                <div>
+                    County
+                    <SelectCustom
+                        optionList={sites}
+                        property={"county"}
+                        value={county}
+                        setValue={setCounty}
+                    />
+                    Town
+                    <SelectCustom
+                        optionList={sites}
+                        property={"town"}
+                        value={town}
+                        setValue={setTown}
+                    />
+                    Site Name
+                    <SelectCustom
+                        optionList={sites}
+                        property={"site_name"}
+                        value={siteName}
+                        setValue={setSiteName}
+                    />
+                    Street<br/>
+                    <input
+                        name="street"
+                        type="text"
+                        value={street}
+                        placeholder="Street Address"
+                        onChange={(e) => setStreet(e.target.value)}
+                    />
+                    <ZipCodeInput
+                        name="Zip Code"
+                        placeholder="Zip Code"
+                        value={zipcode}
+                        onChanged={setZipcode}
+                    />
+                    <FloatInput
+                        name="Latitude"
+                        placeholder="latitude"
+                        value={lat}
+                        onChanged={setLat}
+                    />
+                    <FloatInput
+                        name="Longitude"
+                        placeholder="longitude"
+                        value={long}
+                        onChanged={setLong}
+                    />
```

Comment:
Do these not have names?

Reply:
The "name" property will handle display of a label for these inputs.